File: 974-reorder-data-in-log-files/reorder-data-in-log-files.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        
        letter_logs, digit_logs = [], []

        for log in logs:
            ideti, rest = log.split(' ',1)

            if rest[0].isalpha():
                letter_logs.append((rest,ideti, log))
            else:
                digit_logs.append(log)

        letter_logs.sort(key=lambda x:(x[0],x[1]))

        return [log for ideti, rest, log in letter_logs]+digit_logs

File: 974-reorder-data-in-log-files/reorder-data-in-log-files.java
import java.util.*;

class Solution {
    public String[] reorderLogFiles(String[] logs) {
        List<String[]> letterLogs = new ArrayList<>();
        List<String> digitLogs = new ArrayList<>();
        
        for (String log : logs) {
            int firstSpace = log.indexOf(' ');
            String identi = log.substring(0, firstSpace);
            String rest = log.substring(firstSpace + 1);
            
            if (Character.isLetter(rest.charAt(0))) {
                letterLogs.add(new String[]{rest, identi, log});
            } else {
                digitLogs.add(log);
            }
        }
        
        letterLogs.sort((a, b) -> {
            int cmp = a[0].compareTo(b[0]);
            if (cmp != 0) return cmp;
            return a[1].compareTo(b[1]);
        });
        
        List<String> result = new ArrayList<>();
        for (String[] entry : letterLogs) {
            result.add(entry[2]);
        }
        result.addAll(digitLogs);
        
        return result.toArray(new String[0]);
    }
}

